### DISCUSSION TITLE
Sending lattigo object in a request

### DISCUSSION THREAD

**Discussion by uo257724** [tuneinsight/lattigo#295] [Q&A] [answered]
I am trying to modify the DBFV PIR example, dividing it into a "real" server and clients (3 in this case), and i found the problem of sharing a lattigo object, like when it agregattes shares to create after the pk in server. Is there any way to transform this objects into, for example json, to send them to other member of the party?
Thanks in advance.

**Comment by ChristianMct** [comment, marked as answer]
Hi @uo257724,

All relevant Lattigo object comply to the `BinaryMarshaler` interface. You can use the `obj.MarshalBinary()` and `obj.UnmarshalBinary(data)` function to respectively encode and decode your object into and from `[]byte`.

Note that JSON is not the recommended way for serializing these objects since they can be quite large.

**Reply by uo257724** [reply]
Thanks for the answer,

if i encode the bytes to encode64, will it work better?

**Reply by ChristianMct** [reply]
The `Encode64` methods enable serialization to pre-allocated `[]byte` buffers. It might be more efficient than using the `BinaryMarshaller` interface which re-allocates the buffers on the fly, but it really depends on the application.
